### PR TITLE
ci: Release on edge the latest version every two weeks

### PR DESCRIPTION
**Pull request by locnnil**
If the smoke tests passes, it releases to `beta`.

- Depends on: https://github.com/canonical/docker-snap/pull/199

### CODE REVIEW

File: .github/workflows/upgrade.sh
Comment (farshidtz) [inline .github/workflows/upgrade.sh:10]: This will be very fragile. What if a third such tag gets added?
I suggest relying on the most recent tag using git or [release](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28) instead.

**Review by farshidtz — CHANGES_REQUESTED** [review]

**Comment by farshidtz** [comment]
Thanks for the initiative. 
I support the idea of automating for the upgrade, but the proposed method is a bit problematic as it builds and releases the snap with code that is not checked in, making it hard to audit the artifact against sources. With this release workflow, the changed code never even makes it back to the upstream.

Instead, I propose a semi-automated approach with automated code changes submitted via a PR. This is a well established pattern used by automated dependency update tools such as Dependabot and Renovate. The PR should allow maintainers to perform QA on the changes and the build. Once approved and merged, the remaining automation can take place for the release.

**Comment by locnnil** [comment]
> Instead, I propose a semi-automated approach with automated code changes submitted via a PR. This is a well established pattern used by automated dependency update tools such as Dependabot and Renovate. The PR should allow maintainers to perform QA on the changes and the build. Once approved and merged, the remaining automation can take place for the release.

Sure, I also like that approach.

**Comment by farshidtz** [comment]
Thanks. I'll close this then.

**Pull request closed without merging**
